### DISCUSSION TITLE
Why does setting a default authentication scheme unnecessarily add Authentication middleware to endpoints that don't require authentication?

### DISCUSSION THREAD

**Discussion by a510** [dotnet/aspnetcore#55373] [Q&A]
In the code below the `CookieAuthenticationHandler` fires and tries to authenticate the request although I didn't call `app.UseAuthentication()` or `app.UseAuthorization()` or `RequireAuthorization()`. Why is that and how to avoid this behavior for endpoints that don't require authentication?

``` csharp
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddAuthentication("Cookies").AddCookie();

var app = builder.Build();
app.MapGet("/hello", () => "hello world");       // .AllowAnonymous(); doesn't work as well

app.Run();
```

**Comment by gfoidl** [comment]
From [Enabling authentication in minimal apps](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/security?view=aspnetcore-8.0#enabling-authentication-in-minimal-apps)

> By default, the [WebApplication](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.webapplication) automatically registers the authentication and authorization middlewares if certain authentication and authorization services are enabled.
